libsql-server/src/hrana/ws/session.rs

```diff
@@ -22,6 +22,29 @@ pub struct Session {
     cursors: HashMap<i32, i32>,
 }
 
+impl Session {
+    pub fn new(auth: Authenticated, version: Version) -> Self {
+        Self {
+            auth,
+            version,
+            streams: HashMap::new(),
+            sqls: HashMap::new(),
+            cursors: HashMap::new(),
+        }
+    }
+
+    pub fn update_auth(&mut self, auth: Authenticated) -> Result<(), Error> {
+        if self.version < Version::Hrana2 {
+            bail!(ProtocolError::NotSupported {
+                what: "Repeated hello message",
+                min_version: Version::Hrana2,
+            })
+        }
+        self.auth = auth;
+        Ok(())
+    }
+}
+
```

Comment:
Why is this here? This is already in main -> https://github.com/tursodatabase/libsql/blob/main/libsql-server/src/hrana/ws/session.rs#L26

Could you rebase the change @shopifyski please?